spec/helpers/exclusionCountStepSpec.js

```diff
@@ -0,0 +1,53 @@
+var ExclusionCountStep = require( "../../js/helpers/exclusionCountStep.js" );
+
+describe( "Creates a syllable counter for an exclusion", function() {
+	it( "returns a syllable count step", function() {
+		var mockConfig = {
+			exclusionParts: [
+				{
+					word: "test",
+					syllables: 1
+				}
+			],
+			matchBegin: true,
+			matchEnd: true
+		};
+
+		var step = new ExclusionCountStep( mockConfig );
+		expect( step.createRegex( "test" ) ).toEqual( /(^test|test$)/ );
+		expect( step.countSyllables( "teststring" ).syllableCount ).toBe ( 1 );
+		expect( step.countSyllables( "teststring" ).word ).toBe ( " string" );
+	} );
+
+	it( "returns a syllable count step", function() {
+		var mockConfig = {
+			exclusionParts: [
+				{
+					word: "foo",
+					syllables: 1
+				}
+			],
+			matchBegin: true,
+			regexBeginLetters: [ "s", "t", "n" ]
+		};
+		var step = new ExclusionCountStep( mockConfig );
+
+		expect( step.createRegex( "foo" ) ).toEqual( /(^foos|^foot|^foon)/ );
+		expect( step.countSyllables( "foobar" ).syllableCount ).toBe( 0 );
```

Comment:
Since foo should only be matched if it is followed by an s, t, or n. Since that isn't the case, it returns 0. Will add another spec where it will match.